### PR TITLE
Display tasks with future start date dimly

### PR DESCRIPTION
**Pull request by nico-nacq**
Render future tasks (tasks with a start date in the future) differently than other tasks : Lower opacity, little icon on the right. Italic text.

Before : 

After : 

What changes :

EDIT : After a few tweeks suggested by @raimund-schluessler, here's the result : 

(no italic, better icon and alt text)

(This is the first time I make a PR. Hope I'm doing it right 🤞)

Thank you 😊

### CODE REVIEW

File: src/components/TaskBody.vue [hunk 1]
```diff
@@ -652,6 +653,27 @@ $breakpoint-mobile: 1024px;
 		opacity: .6;
 	}
 
+	&--non-started .title {
+		color: var(--color-placeholder-dark);
+		font-style: italic;
+	}
+
+	&--non-started .title:before {
+		content: '⏱️';
```
Comment (raimund-schluessler) [inline src/components/TaskBody.vue:662]: Please use material design icon for this. Just put it next to the `Notes` and `Pin` icons here:
https://github.com/nextcloud/tasks/blob/00dc26d40aec882196a3cf56bd065d6000532e1f/src/components/TaskBody.vue#L77-L82

Comment (raimund-schluessler) [reply src/components/TaskBody.vue:662]: The clock icon would be suitable, I think:
https://materialdesignicons.com/icon/alarm

File: src/components/TaskBody.vue [hunk 2]
```diff
@@ -652,6 +653,27 @@ $breakpoint-mobile: 1024px;
 		opacity: .6;
 	}
 
+	&--non-started .title {
+		color: var(--color-placeholder-dark);
+		font-style: italic;
```
Comment (raimund-schluessler) [inline src/components/TaskBody.vue:658]: Not so sure about the italic, we don't use it very often in Nextcloud. I would tend to remove it.

**Comment by raimund-schluessler** [comment]
@nico-nacq Thanks for your PR. I left a few comments. In principle I am fine with changing the appearance of future tasks, but I would like to hear @nextcloud/designers about this as well.

File: src/components/TaskBody.vue (hunk 1 above)
Comment (nico-nacq) [reply src/components/TaskBody.vue]: > Please use a material design icon for this. Just put it next to the `Notes` and `Pin` icons

Yes of course : 5243df0 ... Thanks for pointing me the right direction !

> The clock icon would be suitable, I think

If I may, I prefere the `calendar-clock`
image

I think it looks more consistent with the `calendar-start` used in the task settings :
image

It looks like this (with the alt text) :

File: src/components/TaskBody.vue (hunk 2 above)
Comment (nico-nacq) [reply src/components/TaskBody.vue]: Done !

a191d5e

**Comment by nico-nacq** [comment]
> @nico-nacq Thanks for your PR. I left a few comments. In principle I am fine with changing the appearance of future tasks, but I would like to hear @nextcloud/designers about this as well.

Thank you so much for considering my PR and for your work !

File: src/components/TaskBody.vue (hunk 1 above)
Comment (raimund-schluessler) [reply src/components/TaskBody.vue]: > > The clock icon would be suitable, I think
> 
> If I may, I prefere the `calendar-clock` image

Fine with me.

File: src/components/TaskBody.vue [hunk 3]
```diff
@@ -652,6 +656,18 @@ $breakpoint-mobile: 1024px;
...
+	}
+
+	&--non-started > .task-item__body {
+		background-color: var(--color-background-dark);
+	}
+
+	&--non-started .task-checkbox {
+		opacity: .2;
+	}
+
```
Comment (raimund-schluessler) [inline src/components/TaskBody.vue:670]:
```suggestion
	&--non-started {
		.title {
			color: var(--color-placeholder-dark);
		}
		> .task-item__body {
			background-color: var(--color-background-dark);
		}
		.task-checkbox {
			opacity: .2;
		}
	}
```

**Comment by jancborchardt** [comment]
Nice, good idea!

Just some concern regarding accessibility and readability: We do have the variable `--color-text-maxcontrast` which is subdued but ensures proper contrast, but it has to be on `--color-main-background` (not grey as well) to be readable. Would be better to use that, and the text + checkbox color adjustment is probably enough?

**Comment by raimund-schluessler** [comment]
> Nice, good idea!
> 
> Just some concern regarding accessibility and readability: We do have the variable `--color-text-maxcontrast` which is subdued but ensures proper contrast, but it has to be on `--color-main-background` (not grey as well) to be readable. Would be better to use that, and the text + checkbox color adjustment is probably enough?

Would look like this (second task starts in the future, icon omitted, since it's only done in devtools):
Screenshot 2022-05-16 at 14-49-35 Aufgaben - Nextcloud

**Comment by raimund-schluessler** [comment]
I would prefer it without a background as well. If one wants a different background color, it will be very simple to change it with the custom CSS app, since we now set a CSS class for tasks starting in the future with this PR.

**Comment by nico-nacq** [comment]
> Just some concern regarding accessibility and readability: We do have the variable `--color-text-maxcontrast` which is subdued but ensures proper contrast, but it has to be on `--color-main-background` (not grey as well) to be readable. Would be better to use that, and the text + checkbox color adjustment is probably enough?

Ok done : a1abc27

 - The background stays unchanged.
 - The text goes to `--color-text-maxcontrast`.
 - The checkbox still fades to .2
 - The icon is still added on the right.

It looks like this  :

**Comment by nico-nacq** [comment]
Thank you 🙏🥲

**Comment by jancborchardt** [comment]
Looks great, nice work @nico-nacq! Looking forward to more design contributions from you if you like. :) Check https://nextcloud.com/design for more info

**Pull request merged**